RecoLocalMuon/RPCRecHit/src/RPCPointProducer.cc

```diff
@@ -25,44 +25,27 @@
 
 // user include files
 
-RPCPointProducer::RPCPointProducer(const edm::ParameterSet& iConfig)
+RPCPointProducer::RPCPointProducer(const edm::ParameterSet& iConfig) :
+  cscSegments(  consumes<CSCSegmentCollection>(iConfig.getParameter<edm::InputTag>("cscSegments")) ),
+  dt4DSegments( consumes<DTRecSegment4DCollection>(iConfig.getParameter<edm::InputTag>("dt4DSegments")) ),
+  tracks(       consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("tracks")) ),
+  tracks_(      iConfig.getParameter<edm::InputTag>("tracks") ),
+  incldt(       iConfig.getUntrackedParameter<bool>("incldt", true) ),
+  inclcsc(      iConfig.getUntrackedParameter<bool>("inclcsc", true) ),
+  incltrack(    iConfig.getUntrackedParameter<bool>("incltrack", true) ), 
+  debug(        iConfig.getUntrackedParameter<bool>("debug", false) ),
+  MinCosAng(    iConfig.getUntrackedParameter<double>("MinCosAng", 0.95) ),
+  MaxD(         iConfig.getUntrackedParameter<double>("MaxD", 80.) ),
+  MaxDrb4(      iConfig.getUntrackedParameter<double>("MaxDrb4", 150.) ),
+  ExtrapolatedRegion( iConfig.getUntrackedParameter<double>("ExtrapolatedRegion", 0.5) ),
+  trackTransformerParam( iConfig.getParameter<edm::ParameterSet>("TrackTransformer") )
 {
-  cscSegments = consumes<CSCSegmentCollection>(iConfig.getParameter<edm::InputTag>("cscSegments"));
-  
-  dt4DSegments = consumes<DTRecSegment4DCollection>(iConfig.getParameter<edm::InputTag>("dt4DSegments"));
-  tracks = consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("tracks"));
-  tracks_ = iConfig.getParameter<edm::InputTag>("tracks");
-
-  debug=iConfig.getUntrackedParameter<bool>("debug",false);
-  incldt=iConfig.getUntrackedParameter<bool>("incldt",true);
-  inclcsc=iConfig.getUntrackedParameter<bool>("inclcsc",true);
-  incltrack=iConfig.getUntrackedParameter<bool>("incltrack",true);
-  MinCosAng=iConfig.getUntrackedParameter<double>("MinCosAng",0.95);
-  MaxD=iConfig.getUntrackedParameter<double>("MaxD",80.);
-  MaxDrb4=iConfig.getUntrackedParameter<double>("MaxDrb4",150.);
-  ExtrapolatedRegion=iConfig.getUntrackedParameter<double>("ExtrapolatedRegion",0.5);
-
   produces<RPCRecHitCollection>("RPCDTExtrapolatedPoints");
   produces<RPCRecHitCollection>("RPCCSCExtrapolatedPoints");
   produces<RPCRecHitCollection>("RPCTrackExtrapolatedPoints");
-  trackTransformerParam = iConfig.getParameter<edm::ParameterSet>("TrackTransformer");  
 }
 
-
-RPCPointProducer::~RPCPointProducer(){
-
-}
-
-void RPCPointProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
-  /*
-  struct timespec start_time, stop_time;
-  time_t fs;
-  time_t fn;
-  time_t ls;
-  time_t ln;
-  clock_gettime(CLOCK_REALTIME, &start_time);  
-  */
-
+void RPCPointProducer::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
```

Comment:
Best way to do it is to walk yourself through the code in CMSSW, lets you
catch everything but externals with problems and then really complicated
template code like stuff in reco::parser.

helgrind works but it is a bit noisey and things all of the std::atomic<>
variables are causing data races, hard to figure out what exactly is
happening.

On Mon, Aug 24, 2015 at 1:57 AM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoLocalMuon/RPCRecHit/src/RPCPointProducer.cc
> <https://github.com/cms-sw/cmssw/pull/10909#discussion_r37712383>:
>
> > -
> > -RPCPointProducer::~RPCPointProducer(){
> > -
> > -}
> > -
> > -void RPCPointProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
> > -  /*
> > -  struct timespec start_time, stop_time;
> > -  time_t fs;
> > -  time_t fn;
> > -  time_t ls;
> > -  time_t ln;
> > -  clock_gettime(CLOCK_REALTIME, &start_time);
> > -  */
> > -
> > +void RPCPointProducer::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
>
> I know only of running helgrind
> @Dr15Jones <https://github.com/Dr15Jones> may know better
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/10909/files#r37712383>.
>